modules/siddhi-core/src/test/java/io/siddhi/core/query/table/util/TestStoreForCacheMiss.java

```diff
@@ -0,0 +1,309 @@
+        for (Attribute attribute: tableDefinition.getAttributeList()) {
+            testStoreContainingIMTableDefinition.attribute(attribute.getName(), attribute.getType());
+        }
+        for (Annotation annotation: tableDefinition.getAnnotations()) {
+            if (!annotation.getName().equalsIgnoreCase("Store")) {
+                testStoreContainingIMTableDefinition.annotation(annotation);
+            }
+        }
+
+        inMemoryTable.init(testStoreContainingIMTableDefinition, storeEventPool, testTableStreamEventCloner,
+                configReader, super.getSiddhiAppContext(), recordTableHandler);
+    }
+
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+
+    }
+
+    @Override
+    protected RecordIterator<Object[]> query(Map<String, Object> parameterMap, CompiledCondition compiledCondition,
+                                             CompiledSelection compiledSelection, Attribute[] outputAttributes) {
+        StreamEvent outEvent = inMemoryTable.find(compiledCondition, findMatchingEvent);
+        List<Object[]> objects = new LinkedList<>();
+        CompiledSelectionWithCache compiledSelectionWithCache = null;
+
+        if (outEvent != null) {
+            compiledSelectionWithCache = (CompiledSelectionWithCache) compiledSelection;
+            StateEventFactory stateEventFactory = new StateEventFactory(compiledSelectionWithCache.getMetaStateEvent());
+            Event[] cacheResultsAfterSelection = executeSelector(outEvent,
+                    compiledSelectionWithCache.getQuerySelector(), stateEventFactory, MetaStreamEvent.EventType.TABLE);
+
+            if (compiledSelectionWithCache.getQuerySelector() !=  null &
+                    compiledSelectionWithCache.getQuerySelector().getAttributeProcessorList().size() !=  0) {
+                compiledSelectionWithCache.getQuerySelector().
+                        process(generateResetComplexEventChunk(outEvent.getOutputData().length, stateEventFactory));
+            }
+            if (cacheResultsAfterSelection != null) {
+                for (Event event: cacheResultsAfterSelection) {
+                    objects.add(event.getData());
+                }
+            }
+        }
+        return new TestStoreWithCacheIterator(objects.iterator());
+    }
+
+    @Override
+    protected CompiledSelection compileSelection(List<SelectAttributeBuilder> selectAttributeBuilders,
+                                                 List<ExpressionBuilder> groupByExpressionBuilder,
+                                                 ExpressionBuilder havingExpressionBuilder,
+                                                 List<OrderByAttributeBuilder> orderByAttributeBuilders, Long limit,
+                                                 Long offset) {
+        CompiledSelectionWithCache compiledSelectionWithCache;
+        MetaStateEvent metaStateEvent = matchingMetaInfoHolderForTestStoreQuery.getMetaStateEvent().clone();
```

Comment:
If we are using clone in only this place lets remove the clone and move the logic here itself.

Reply:
I needed to clone MSE on the previous occasion as well. But since the method was not available I created MSE by debugging. If clone function is available it will make future developments easier